### PR TITLE
Adds flag AMBIENT_LIGHT_DISABLED to GLES2 

### LINKED ISSUE #28171: GLES2: Flag "Disable Ambient Light" in spatial material doesn't work
**Godot version:**
Godot 3.1 stable

**OS/device including version:**
Zorin OS core 12.4

**Issue description:**
The Disable Ambient Light flag in the spatial material doesn't work.

**Steps to reproduce:**
- Create a material
- Check Disable Ambient Light to true
- Nothing happens

Posible related: #27061

### PR DESCRIPTION
**Pull request by SouzaGuilherme**
Closes #28171.

### CODE REVIEW

File: drivers/gles2/shaders/scene.glsl [hunk 1]
```diff
@@ -1542,6 +1542,11 @@ FRAGMENT_SHADER_CODE
 
 #ifdef BASE_PASS
 	//none
+
+#ifdef AMBIENT_LIGHT_DISABLED
+    ambient_light = vec3(0.0, 0.0, 0.0);
```
Comment (akien-mga) [inline drivers/gles2/shaders/scene.glsl]: Style issue, indentation should use a tab. There's also a typo in the commit log, it should be GL**E**S2.

File: drivers/gles2/shaders/scene.glsl [hunk 2]
```diff
@@ -1564,6 +1569,7 @@ FRAGMENT_SHADER_CODE
 
 #endif
 
+#endif
```
Comment (akien-mga) [inline drivers/gles2/shaders/scene.glsl]: Could you add `// AMBIENT_LIGHT_DISABLED` to make the scope clear?

File: drivers/gles2/shaders/scene.glsl (hunk 1 above)
Comment (SouzaGuilherme) [reply drivers/gles2/shaders/scene.glsl]: Thank you, I will fix it accordingly.

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.1.2.

**Pull request merged**